### DISCUSSION TITLE
parse day off by one

### DISCUSSION THREAD

**Discussion by rhufsky** [js-fns/date-fns#2504] [Q&A] [answered]
I have node 14.17.0
date-fns 2.22.1

I parse a new date object from a string that says 24.9.1960 in various formats. 
I constantly get a date object that says 23.9.1960.

What can I do to fix this?

```
import { parse } from "date-fns";

//const aDate = parse("24.09.1960", "dd.MM.yyyy", new Date());
//const aDate = parse("24-09-1960", "dd-MM-yyyy", new Date());
const aDate = parse("1960-09-24", "yyyy-MM-dd", new Date());

console.log(aDate);
```

**Comment by fturmel** [comment, marked as answer]
The input is treated as being in the system's local time zone, not as UTC.

When you `console.log` a Date object in Node, it stringifies it using `.toISOString()` which is a UTC interpretation of the date. So, if you are in a time zone with a positive offset related to UTC, any midnight local date will be represented as the previous day in UTC.

If you really meant that input to be UTC, you could use the `X` token and add a `Z` in the date string.

```js
// Note: These are outputs from an eastern time zone system

// same as: new Date(1960, 8, 24)
let d = parse('1960-09-24', 'yyyy-MM-dd', new Date());
console.log(d.toString()); // Sat Sep 24 1960 00:00:00 GMT-0400 (Eastern Daylight Time)
console.log(d.toISOString()); // 1960-09-24T04:00:00.000Z

// same as: new Date(Date.UTC(1960, 8, 24))
d = parse('1960-09-24Z', 'yyyy-MM-ddX', new Date());
console.log(d.toString()); // Fri Sep 23 1960 20:00:00 GMT-0400 (Eastern Daylight Time)
console.log(d.toISOString()); // 1960-09-24T00:00:00.000Z
```

It's worth mentioning that `date-fns` doesn't have UTC variants of the functions at this point (see #376), so depending on what type of date operations you might be doing after parsing, you may want to keep it in the local time zone throughout or consider using the `date-fns-tz` library.

This may also be useful: https://date-fns.org/docs/Time-Zones

**Reply by rhufsky** [reply]
Thanks a lot, that clarified it for me!
